third_party/move/move-compiler-v2/src/logging.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! Configures the logging backend for the compiler and related code which uses
+//! the `log` crate (macros `info!`, `error!`, etc.).
+//!
+//! # How to Run the Logger
+//!
+//! By default logging is turned off. The environment variable `MVC_LOG` is
+//! used to control logging. Usages:
+//!
+//! ```
+//!     MVC_LOG = ""                # logs everything to console (stderr)
+//!     MVC_LOG = "info"            # logs only info or higher to console
+//!     MVC_LOG = "<module_path_prefix>=info" # logs only info+ in matching modules
+//!     MVC_LOG = ":<file>"         # logs everything to given file
+//!     MVC_LOG = "info:<file>      # as above
+//! ```
+//!
+//! The general format is `<spec>[:<file>]`, where `spec` is defined as described
+//! for the `LogSpecification` type.
+//!
+//! # How to write Logs
+//!
+//! One can import and use the following macros in decreasing severity: `error!`, `warn!`, `info!`,
+//!`debug!`, and `trace!`. Also any other code in external crates already using `log` will
+//! use those macros. Invocations of those macros will be redirected to the logger configured
+//! via the `MVC_LOG` environment variable.
+//!
+//! Those macros expand to a conditional, and it is a quick check and branch to skip
+//! logging if the level is not enabled (which is the default). In addition, one can
+//! use the boolean macro `log_enabled!(log::Level::Debug)` to check whether a given
+//! level is enabled.
+//!
+//! In general it is good to keep INFO and higher severity log messages in the code. They
+//! may help to debug customer issues in production. However, for keeping the code base clean,
+//! uses of the `debug!` and lower logging level should be kept minimal and possibly removed
+//! eliminated. Its a judgement call where and for how long to leave them in code.
+
+use flexi_logger::{DeferredNow, FileSpec, LogSpecification, Logger};
+use log::Record;
+use std::env;
+
+const MVC_LOG_VAR: &str = "MVC_LOG";
```

Comment:
For a couple of minutes, I was wondering how model-view-controller (MVC) relates here (as far as the naming of the env var `MVC_LOG` goes). But I realized it might stand for for "move compiler log"?

In either case, might be worth adding a brief comment with the intended full expansion somewhere (either when the variable is introduced first in the module comments, or here).

Reply:
I'm planning to suggest that we use `MVC` where ever we use `MOVE_COMPILER` in variables right now. Forgot to mention in the thread.